packages/core/src/index.ts

```diff
@@ -427,7 +427,18 @@ export async function createFileWatcher(
 
       await devServer.close();
       __FARM_GLOBAL__.__FARM_RESTART_DEV_SERVER__ = true;
-      await start(resolvedUserConfig as FarmCLIOptions & UserConfig);
+
+      const defaultOptions = {
+        root: resolvedUserConfig.root,
+        compilation: {
+          lazyCompilation: resolvedUserConfig.compilation.lazyCompilation
+        },
+        server: resolvedUserConfig.server,
+        clearScreen: resolvedUserConfig.clearScreen,
+        configPath: path.dirname(resolvedUserConfig.configFilePath),
```

Comment:
Is there any difference between this and passing resolveConfig directly? I think only the path has been modified. Then we should adjust the configPath issue when parsing the config.

Reply:
I think maybe these designs are all problematic. @wre232114  take a look ？

Reply:
> I think maybe these designs are all problematic. @wre232114 take a look ？

I have a similar feeling that perhaps we need to rewrite the configuration part somewhat in the future, but I think this solution is the best we can do without a major rewrite to avoid these issues.

Reply:
A pure restart is safe and causes no more problems, and you don't seem to realize that if we remove a configuration instead of overwriting it with a configuration, it will use the old value instead of the default.

Reply:
I just remove this change, maybe we should fix this in #1589